From 6197e8d0232b870948b77193dba770f6c2702199 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:01 +0800
Subject: [PATCH 020/150] dt-bindings: can: phytium: Add bindings for Phytium
 CAN

This patch documents the DT bindings for Phytium CAN controller.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: I12c22d3fd5d0b70bb88bcf70eafcfddda84f8c4f
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../bindings/net/can/phytium,can.yaml         | 72 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 73 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/can/phytium,can.yaml

diff --git a/Documentation/devicetree/bindings/net/can/phytium,can.yaml b/Documentation/devicetree/bindings/net/can/phytium,can.yaml
new file mode 100644
index 000000000000..dd4b8000830c
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/can/phytium,can.yaml
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/can/phytium,can.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title:
+  Phytium CAN controller
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    enum:
+      - phytium,can
+      - phytium,canfd
+
+  reg:
+    maxItems: 1
+    description: Should contain the controller registers location and length
+
+  interrupts:
+    maxItems: 1
+    description: Should contain IRQ line for the controller
+
+  clocks:
+    maxItems: 1
+    description: Input clock used by the controller
+
+  clock-names:
+    maxItems: 1
+    description: Input clock name, should be "can_clk"
+    const: can_clk
+
+  tx-fifo-depth:
+    $ref: "/schemas/types.yaml#/definitions/uint32"
+    description: Indicates the length of TX FIFO
+
+  rx-fifo-depth:
+    $ref: "/schemas/types.yaml#/definitions/uint32"
+    description: Indicates the length of RX FIFO
+
+  extend_brp:
+    description: |
+      Indicates to apply the extend BRP parameter of bit timming for
+      early version of CAN controller
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - clocks
+  - clock-names
+  - tx-fifo-depth
+  - rx-fifo-depth
+
+allOf:
+  - $ref: can-controller.yaml#
+
+examples:
+  - |
+
+    can0: can@2800a000 {
+      compatible = "phytium,canfd";
+      reg = <0x0 0x2800a000 0x0 0x1000>;
+      interrupts = <GIC_SPI 81 IRQ_TYPE_LEVEL_HIGH>;
+      clocks = <&sysclk_600mhz>;
+      clock-names = "can_clk";
+      tx-fifo-depth = <64>;
+      rx-fifo-depth = <64>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 2e81b1bc508e..492b97e22477 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2532,6 +2532,7 @@ S:	Maintained
 W:	https://www.phytium.com.cn
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,sdci.yaml
+F:	Documentation/devicetree/bindings/net/can/phytium,can.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,qspi-nor.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,spi.yaml
 F:	arch/arm64/boot/dts/phytium/*
-- 
2.47.0

